docs/roadmap.md

```diff
@@ -4,6 +4,149 @@ title: Babel Roadmap
+
+### Continue developing the `babel-polyfills` project
+
+We have already decided to remove the old `core-js@2` support from `@babel/preset-env` in Babel 8, and we also want to stop promoting a specific third-party polyfill giving our users the impression that it's part of Babel itself.
+
+This might happen in two different ways:
+- We just remove `core-js@3` from `@babel/preset-env` in Babel 8, encouraging users to migrate to `babel-plugin-polyfill-corejs3` (which is what `@babel/preset-env` internally uses starting from version 7.10.0)
+- We can keep `core-js@3` support in `@babel/preset-env`, but not migrate it to `@babel/core` when we'll move the transform plugins.
+
+Whatever path we take, we would like to offer at least one alternative to our users when they'll need to update the `core-js` integration in their configuration. `core-js` is a really good polyfill that ensures the highest possible spec compliancy, but different users might prefer different trade-offs.
+
+([Nicolò](https://github.com/nicolo-ribaudo)) is working with [@ljharb](https://github.com/ljharb) to make sure that the [`@es-shims` project](https://github.com/es-shims/) supports at least all the ES2015+ features (we actually aim for ES5+), so that Babel users are free to choose to between at least two alternatives.
+
+This needs to happen _before_ dropping built-in support for `core-js@3`, so that people interested in `es-shims` don't have to migrate twice.
+
+### Expand `targets` usage for granular transforms
+
+Since the beginning, `@babel/preset-env` has used the `targets` option to automatically enable or disable transform plugins.
+
+However, there isn't a 1-to-1 mapping between Babel plugins and features implemented in browsers.
+
+For example, we have a single plugin for the different class fields types (public and private, static and instance), but browsers have varying compatibility matrices:
+
+- Firefox 73 and Safari 14 support only public instance fields
+- Firefox 75+ supports pulic instance and static fields
+- Chrome 79+ supports public and private fields, but doesn't support private fields in some optional chaining expressions
+- Chrome 84+ fully supports private fields, and also private methods
+- Safari TP 121 fully supports private fields (even with `?.`), but it doesn't support private methods
+
+Creating a plugin for each feature is sub-optimal. For example, we can convert private methods to private fields and then, if needed, convert them to older syntax. However, we can generate a better optimized output by directly converting private methods down to older syntax without the intermediate step, if we know that it needs to be transpiled down.
+
+Since starting from Babel 7.13.0 we can read `targets` option directly inside a plugin, we can modify our plugins to automatically perform _partial_ compilation of a given ECMAScript feature, where this would give an advantage in the output size and runtime performance.
```

Comment:
```suggestion
Since Babel 7.13.0, we can read the `targets` option directly inside a plugin, we can modify our plugins to automatically perform _partial_ compilation of a given ECMAScript feature, where this would give an advantage in the output size and runtime performance.
```